src/NUnitFramework/framework/Api/FrameworkController.cs

```diff
@@ -558,7 +553,7 @@ public class StopRunAction : FrameworkControllerAction
             /// <remarks>A forced stop will cause threads and processes to be killed as needed.</remarks>
             public StopRunAction(FrameworkController controller, bool force, object handler)
             {
-                controller.StopRun((ICallbackEventHandler)handler, force);
+                controller.StopRun(force);
```

Comment:
These changes won't change current behavior, though the mere existence of the `object handler` parameter makes me wonder if there's a bug in the current behavior.

Reply:
> wonder if there's a bug in the current behavior

yeah i thought the same. but i lack the history so if there is a bug, i could not see it

Reply:
I got curious and dug into this. It looks like `FrameworkController.StopRun()` would at one point raise a callback event
https://github.com/nunit/nunit/commit/362f7a3f7074ff520ad811ba06883b3595aa0685#diff-a4de5abf6ba13245db808314f98df69cd318b068215f50932b0d50abdd938c43R203
```csharp
private void StopRun(ICallbackEventHandler handler, bool force, int timeout)
        {
            //handler.RaiseCallbackEvent(Runner.StopRun(force, timeout).ToString());
            Runner.StopRun(force);
            handler.RaiseCallbackEvent(Runner.WaitForCompletion(timeout).ToString());
        }
```
It looks like it was removed at some point many (> 7) years ago though I haven't dug further to an exact commit

Reply:
ok i reverted this part of the PR given it needs more work